trajectory_controller:
  ros__parameters:

    # lookahead: 0.05
    # velocity: 0.2
    goal_position_tolerance: 0.1
    goal_heading_tolerance: 0.1
    lookahead: 0.2
    max_velocity: 0.5
    curvature_gain: 2.0
    k_theta: 0.8
    omega_max: 1.8

    k1: 0.8
    k2: 4.5
    k3: 2.2
    # waypoints: [                                        #straight line
    #   0.0, 0.0,
    #   5.0, 0.0
    # ]

    # waypoints: [                                          # gentle_curve
    #   0.0, 0.0,
    #   2.0, 0.0,
    #   2.0, 2.0
    # ]
    
    # waypoints: [                                            # sharp turn
    #   0.0, 0.0,
    #   2.0, 0.0,
    #   2.0, 2.0
    # ]

    waypoints: [

      # --- Straight ---
      0.0, 0.0,
      2.0, 0.0,

      # --- 90 degree turn ---
      2.0, 2.0,

      # --- Zig-zag ---
      3.0, 2.5,
      4.0, 1.5,
      5.0, 2.5,
      6.0, 1.5,

      # --- S-curve ---
      5.5, 0.5,
      4.0, -1.0,
      2.5, 0.5,

      # --- Tight U-turn ---
      1.5, 2.0,
      0.5, 1.5,
      1.0, 0.0,

      # --- Final smooth return ---
      2.5, -1.5,
      4.5, -1.0,
      6.0, 0.0

    ]
